indigo_app/views/places.py

```diff
@@ -144,3 +146,61 @@ def get_context_data(self, **kwargs):
         })
 
         return context
+
+    def coalesce_entries(self, stream):
+        """ If more than 1 task were added to a workflow at once, rather display something like
+        '<User> added <n> tasks to <workflow> at <time>'
+        """
+        activity_stream = []
+        added_stash = []
+        for i, action in enumerate(stream):
+            if i == 0:
+                # is the first action an addition?
+                if getattr(action, 'verb', None) == 'added':
+                    added_stash.append(action)
+                else:
+                    activity_stream.append(action)
+
+            else:
+                # is a subsequent action an addition?
+                if getattr(action, 'verb', None) == 'added':
+                    # if yes, was the previous action also an addition?
+                    prev = stream[i - 1]
+                    if getattr(prev, 'verb', None) == 'added':
+                        # if yes, did the two actions happen close together and was it on the same workflow?
+                        if prev.timestamp - action.timestamp < self.threshold \
+                                and action.target_object_id == prev.target_object_id:
+                            # if yes, the previous action was added to the stash and
+                            # this action should also be added to the stash
+                            added_stash.append(action)
```

Comment:
The logic here is fine. It's worth noting that all of these blocks end with `added_stash.append(action)`. That means you could probably remove a bunch of `else` statements and just have `added_stash.append(action)` at the end of the if block that starts on line 169.